### PR TITLE
autopilot_upgrade_version should be parseable

### PR DESCRIPTION
**Pull request by raskchanky**
I forgot to include this check in the original PR. I'm assigning the nochangelog label because this technically falls under the enterprise autopilot changelog.

### CODE REVIEW

File: physical/raft/raft.go [hunk 1]
```diff
@@ -459,6 +460,15 @@ func NewRaftBackend(conf map[string]string, logger log.Logger) (physical.Backend
 		return nil, fmt.Errorf("autopilot_reconcile_interval (%v) should be larger than autopilot_update_interval (%v)", effectiveReconcileInterval, effectiveUpdateInterval)
 	}
 
+	var upgradeVersion string
+	if uv, ok := conf["autopilot_upgrade_version"]; ok && uv != "" {
+		upgradeVersion = uv
+		_, err := goversion.NewVersion(upgradeVersion)
```
[the remark this answers is not in the document]
Comment (raskchanky) [reply physical/raft/raft.go:466]: Yes

[the remark this answers is not in the document]
Comment (raskchanky) [reply physical/raft/raft.go:466]: My thought was it's probably better to fail sooner rather than later.

[the remark this answers is not in the document]
Comment (ncabatoff) [reply physical/raft/raft.go:466]: Sounds good, please make sure to note that in the docs.

**Review by HridoyRoy — APPROVED** [review]
Looks good to me, pending @ncabatoff 's question!

**Pull request merged**